### PR TITLE
DROOLS-4725: Cannot validate list in legacy Test Scenario

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-4725

Part of an ensemble:
- https://github.com/kiegroup/drools/pull/2635
- https://github.com/kiegroup/drools-wb/pull/1259

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-test-scenarios/src/test/java/org/drools/workbench/models/testscenarios/backend/verifiers/FactFieldValueVerifierTest.java [hunk 1]
```diff
@@ -54,4 +54,29 @@ public void testSQLDate() throws Exception {
 
         assertTrue(verifyField.getSuccessResult());
     }
+
+    public void testExpectedList() throws Exception {
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-test-scenarios/src/test/java/org/drools/workbench/models/testscenarios/backend/verifiers/FactFieldValueVerifierTest.java:61]: Could we use `@Test` annotation, ideally in whole class.

Comment (manstis) [reply drools-workbench-models/drools-workbench-models-test-scenarios/src/test/java/org/drools/workbench/models/testscenarios/backend/verifiers/FactFieldValueVerifierTest.java:61]: Done.

**Review by jomarko — COMMENTED** [review]
@manstis I would like to increase the coverage https://github.com/manstis/drools/pull/3

**Comment by manstis** [comment]
@jomarko I've merged your PR into this. Thank-you.

**Comment by jomarko** [comment]
@manstis I can confirm your fix works if I import the project provided in jira and update the scenario to have format **=["value1", "value2"]**, however if I try to create similar test scenario from scratch, I am not able to do it. 

See the [project](https://issues.jboss.org/secure/attachment/12457209/MySpace_uhcSearchTest.zip), I prepared **stopWrok.rdrl** and **stopWorkTest.scenario** there. The scenario is not finished. The issue is when I try to finish it like in this [video](https://issues.jboss.org/secure/attachment/12457210/list-test-scenario-legacy.webm), the value I want to check always disappear. may you have a look please?

**Comment by manstis** [comment]
Thank-you @jomarko 

I've had to submit a fix for `drools-wb` too: https://github.com/kiegroup/drools-wb/pull/1259

The problem was that when creating a _new_ expectation the _nature_ of the expectation was not set to a `FieldData.TYPE_LITERAL` and hence when the widget was re-loaded the screen inferred the _nature_ from the value. Since it starts with `=` it was assumed to be `FieldData.TYPE_VARIABLE` and the drop-down shown (with empty content as there were no bound variables). The expectations _nature_ controls what widget is shown for the value.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by jomarko** [comment]
@manstis building manually right now, but the changes doesn't seem to reflect your explanation.

**Review by jomarko — APPROVED** [review]
@manstis my apologizes, changes are correct, just took me a while to understand.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Pull request merged**